Plugins/NativeInput/Source/NativeInput.cpp

```diff
@@ -98,26 +162,31 @@ namespace Babylon::Plugins
                 }
             }
 
-            RemoveInputMap(DeviceType::Touch, pointerId);
-
+            RemoveInputMap(deviceType, pointerId);
             m_eventDispatcher.tick(arcana::cancellation::none());
         });
     }
 
-    void NativeInput::Impl::PointerMove(uint32_t pointerId, uint32_t x, uint32_t y)
+    void NativeInput::Impl::PointerMove(uint32_t pointerId, uint32_t x, uint32_t y, DeviceType deviceType)
     {
-        m_runtimeScheduler([pointerId, x, y, this]() {
-            std::vector<int32_t>& deviceInputs{GetOrCreateInputMap(DeviceType::Touch, pointerId, { POINTER_X_INPUT_INDEX, POINTER_Y_INPUT_INDEX })};
-            SetInputState(DeviceType::Touch, pointerId, POINTER_X_INPUT_INDEX, x, deviceInputs, true);
-            SetInputState(DeviceType::Touch, pointerId, POINTER_Y_INPUT_INDEX, y, deviceInputs, true);
+        m_runtimeScheduler([pointerId, x, y, deviceType, this]() {
+            std::vector<int32_t>& deviceInputs{ GetOrCreateInputMap(deviceType, pointerId, { POINTER_X_INPUT_INDEX, POINTER_Y_INPUT_INDEX }) };
+            SetInputState(deviceType, pointerId, POINTER_X_INPUT_INDEX, x, deviceInputs, true);
+            SetInputState(deviceType, pointerId, POINTER_Y_INPUT_INDEX, y, deviceInputs, true);
 
             m_eventDispatcher.tick(arcana::cancellation::none());
         });
     }
 
     NativeInput::Impl::DeviceStatusChangedCallbackTicket NativeInput::Impl::AddDeviceConnectedCallback(NativeInput::Impl::DeviceStatusChangedCallback&& callback)
     {
-        return m_deviceConnectedCallbacks.insert(std::move(callback));
+        // Firing this callback on registration to mimic web behavior
+        for (auto [key, state] : m_inputs)
+        {
+            callback(key.first, key.second);
```

Comment:
@ryantrem synchronous calls seem to work for this callback, but do we need to actually queue things with the event dispatcher?

Reply:
No we shouldn't need to dispatch. This is called from the JS thread so we are already on the correct thread, and the equivalent code that runs in the browser does this synchronously as well: https://github.com/BabylonJS/Babylon.js/blob/master/src/DeviceInput/deviceInputSystem.ts#L29-L40